Migrate to Java 25

duanemay opened the pull request:
- Replace unused variables with underscore
- Migrate to IO utility class
- Use switch expression

Mostly done with OpenRewrite Rules, and some manual touchups.

duanemay committed:
Migrate to Java 25

- Replace unused variables with underscore
- Migrate to IO utility class
- Use switch expression

duanemay committed:
Merge branch 'develop' into java25

strehle committed:
Merge branch 'develop' into java25

strehle commented on .github/workflows/codeql-analysis.yml:
```diff
@@ -52,7 +52,7 @@ jobs:
       with:
         distribution: 'liberica'
-        java-version: '21'
+        java-version: '25'
```
do you plan to upgrade java within the normal patches or with a major version update ?

duanemay replied:
No real functional changes, so it could be a minor bump

strehle replied:
On one side correct, on the other side it will affect our clients.

With this change the code is only compatible with java 25. As mentioned... we have an internal UAA repo where we build UAA from code and then colleagues needs to be informed to adopt to java 25 because with this PR it would not compile anymore.

duanemay replied:
Good point. Yes, for those consuming from source, this would be an issue. It looks like we did do a major bump to v78 on the Java 21 changes, so should also do a major bump when we decide to merge this.

strehle replied:
@adrianhoelzl-sap FYI

cezary13k commented:
> ## Pull request overview @cezary13k
> 
> This pull request migrates the codebase from Java 21 to Java 25, implementing several modern Java features:
> - Replaces unused exception variables with underscore `_` (JEP 443 - Unnamed Patterns and Variables)
> - Migrates from `System.out.println()` to `IO.println()` utility class
> - Converts traditional switch statements to switch expressions
> - Replaces `.get(0)` with `.getFirst()` for better semantics
> - Updates Java version configuration across build and CI/CD files
> 
> ### Reviewed changes
> 
> Copilot reviewed 249 out of 250 changed files in this pull request and generated 2 comments.
> 
> <details>
> <summary>Show a summary per file</summary>
> 
> | File | Description |
> | ---- | ----------- |
> | build.gradle | Updates Java source and target compatibility from 21 to 25 |
> | .github/workflows/*.yml | Updates CI/CD pipelines to use Java 25 |
> | Test files (multiple) | Replaces unused exception variables with `_`, migrates to `IO.println()` |
> | Main source files (multiple) | Same modernization patterns as test files |
> | Switch statement files | Converts to switch expressions in SecurityFilterChainPostProcessor, SimpleSearchQueryConverter, StringUtilities |
> | Collection access files | Replaces `.get(0)` with `.getFirst()` |
> </details>
> 
> 
> 
> 
> 
> 
> ---
> 
> 💡 <a href="/cloudfoundry/uaa/new/develop/.github/instructions?filename=*.instructions.md" class="Link--inTextBlock" target="_blank" rel="noopener noreferrer">Add Copilot custom instructions</a> for smarter, more guided reviews. <a href="https://docs.github.com/en/copilot/customizing-copilot/adding-repository-custom-instructions-for-github-copilot" class="Link--inTextBlock" target="_blank" rel="noopener noreferrer">Learn how to get started</a>.

duanemay commented:
Waiting on:
- ✅ java buildpack support
- ✅ Bouncy Castle support

duanemay committed:
Migrate code to Java 25 Syntax

Rewrite Rules applied
- Replace unused variables with underscore
- Migrate to IO utility class
- Use switch expression

duanemay committed:
Fix broken logging test

duanemay committed:
Migrate test code to Java 25 Syntax

Rewrite Rules applied

duanemay committed:
Sonar Fixes

duanemay committed:
Assertj Rewrite

duanemay committed:
Assertj Rewrite

Replace assertThatExceptionOfType with assertThatThrownBy

duanemay committed:
Adjust visibility for testing
